### PR TITLE
*: remove old placement code for exchange partition

### LINKED ISSUE #29648: Remove old placement code for `exchange partition`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #29648

remove old placement code for exchange partition

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/partition.go [hunk 1]
```diff
@@ -1467,6 +1458,44 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
...
+	if ptBundle != nil {
+		bundles = append(bundles, ptBundle)
+	}
+
+	parBundle, err := newBundleFromPartitionDef(t, job, *newPar)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if parBundle != nil {
+		bundles = append(bundles, ptBundle)
```
Comment (xhebox) [inline ddl/partition.go:1477]: `ParBundle`?

Comment (lcwangchao) [reply ddl/partition.go:1477]: Yes, it's a mistake

**Review by xhebox — APPROVED** [review]
LGTM
Maybe add a comment for `placement.NewBundle(xxx)`, since `newPar.ID` is actually the id of table...

**Review by AilinKid — APPROVED** [review]

**Pull request merged**